### PR TITLE
fix(ignore_upgrade_schema_errors): fix db_event for cdc issue

### PR DESCRIPTION
**Pull request by dkropachev**
invalid_request_exception, is not considered to be critical - https://github.com/scylladb/scylla/issues/7817

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by fruch** [comment]
@dkropachev why no doing this specifically for the relevant test ?, i.e. the rolling upgrade test ?

**Comment by dkropachev** [comment]
> relevant

I can't be sure about conditions under which this exception is thrown.

**Comment by fruch** [comment]
> > @dkropachev why no doing this specifically for the relevant test ?, i.e. the rolling upgrade test ?
> 
> @fruch I can't be sure about conditions under which this exception is thrown.

according to scylladb/scylla#7817, this happens only when upgrading from 4.2 (with experimental flag) to 4.3.
I.e. those should be ignore only during the upgrade of a node

see https://github.com/scylladb/scylla-cluster-tests/blob/30235790040d8cfd70bb154bc907e73985c2ef25/upgrade_test.py#L491 for example of how we filter error, with the use of a context manager

**Comment by fruch** [comment]
> > > @dkropachev why no doing this specifically for the relevant test ?, i.e. the rolling upgrade test ?
> > 
> > 
> > @fruch I can't be sure about conditions under which this exception is thrown.
> 
> according to [scylladb/scylla#7817](https://github.com/scylladb/scylla/issues/7817), this happens only when upgrading from 4.2 (with experimental flag) to 4.3.
> I.e. those should be ignore only during the upgrade of a node
> 
> see
> 
> https://github.com/scylladb/scylla-cluster-tests/blob/30235790040d8cfd70bb154bc907e73985c2ef25/upgrade_test.py#L491
> 
> for example of how we filter error, with the use of a context manager

also see like we are handling this already:

https://github.com/scylladb/scylla-cluster-tests/blob/30235790040d8cfd70bb154bc907e73985c2ef25/sdcm/sct_events/group_common_events.py#L91

@dkropachev can you explain where you are still seeing this issue ?

**Comment by dkropachev** [comment]
> with ignore_upgrade_schema_errors():

Thanks Israel for pointing me into the issue, PR got changed, take a look

File: unit_tests/test_cluster.py [hunk 1]
```diff
@@ -115,6 +116,16 @@ def test_search_system_interlace_reactor_stall(self):
             assert event_b["type"] == "REACTOR_STALLED"
             assert event_b["line_number"] == 3
 
+    def test_search_cdc_invalid_request(self):
+        from sdcm.sct_events.group_common_events import ignore_upgrade_schema_errors
+        self.node.system_log = os.path.join(os.path.dirname(__file__), 'test_data', 'system_cdc_invalid_request.log')
```
Comment (fruch) [inline unit_tests/test_cluster.py:121]: Why the import is here, is there a cyclic dependency issue here ? If so please comment there is

File: sdcm/sct_events/events_device.py [hunk 2]
```diff
@@ -171,7 +169,12 @@ def outbound_events(self,
                         LOGGER.debug("%s: add filter %s with uuid=%s", self, obj, obj.uuid)
                         filters[obj.uuid] = obj
 
-                if obj_filtered or isinstance(obj, SystemEvent):
+                if isinstance(obj, SystemEvent):
```
Comment (fruch) [inline sdcm/sct_events/events_device.py:172]: I'm not following the reason for this change there was a logic issue ? Or what is exactly the reason for this change ?

File: unit_tests/test_cluster.py (hunk 1 above)
Comment (dkropachev) [reply unit_tests/test_cluster.py:121]: fixed

File: sdcm/sct_events/events_device.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/sct_events/events_device.py:172]: This one is part of the commit "fix(event_device): stop evaluating event filters against event filters"

Comment (dkropachev) [reply sdcm/sct_events/events_device.py:172]: As you can see the purpose is to stop evaluating filters agaist filters.

**Comment by fruch** [comment]
can put a small log into `unit_tests/test_data/system_cdc_invalid_request.log` ? no need for 60K lines for this test

**Comment by dkropachev** [comment]
> can put a small log into `unit_tests/test_data/system_cdc_invalid_request.log` ? no need for 60K lines for this test

Done.

**Review by fruch — APPROVED** [review]
LGTM

**Comment by roydahan** [comment]
@fruch / @dkropachev , I didn't want to invest time on fixing it because now when we release 4.3 and CDC is GA, the upgrade from 4.3 to 4.4 should work without this workaround.
So, shall we revert it now?

**Comment by fruch** [comment]
After we change upgrade to 4.3

**Comment by roydahan** [comment]
We can revert now, because I just merged today the base version to be 4.3

**Pull request merged**